lms/services/email_preferences.py

```diff
@@ -60,6 +74,24 @@ def h_userid(self, url: str) -> str:
         except (ExpiredJWTError, InvalidJWTError) as err:
             raise InvalidTokenError() from err
 
+    def get_preferences(self, h_userid):
+        preferences = self._user_preferences_service.get(h_userid)
+
+        for key in self.DAY_KEYS:
+            preferences.preferences.setdefault(key, True)
+
+        return {key: preferences.preferences[key] for key in self.DAY_KEYS}
+
+    def set_preferences(self, h_userid, new_preferences):
+        self._user_preferences_service.set(
+            h_userid,
+            {
+                key: new_preferences[key]
+                for key in new_preferences
+                if key in self.DAY_KEYS
+            },
```

Comment:
Probably not strictly necessary, but this filters out any non-email preferences in the given `new_preferences`. Other than that this method is just a wrapper for `UserPreferencesService.set()`, it's just convenient to have this wrapper here so that views don't need to use both services.